studio.py

```diff
@@ -14,14 +28,182 @@
+
+            with PROXYS[self.proxy](
+                path=site["recording_path"], mode="record"
+            ) as proxy_service:
+                app_service = APPS[self.app](proxy_service, self.certutil, self.binary)
+                print("Recording %s..." %site["url"])
+                app_service.start(site["url"])
+
+                if not site.get("login", None):
+                    print("Waiting %s for the site to load..." % RECORD_TIMEOUT)
+                    time.sleep(RECORD_TIMEOUT)
+                else:
+                    time.sleep(5)
+                    raw_input("Do user input and press <Return>")
+
+                app_service.screen_shot(site["screen_shot_path"])
+                self.information["app_info"] = app_service.app_information()
+                app_service.stop()
+
+            self.update_json_information(site)
+            self.generate_zip_file(site)
+            self.generate_manifest_file(site)
+
+    def parse_sites_json(self):
+        print("Parsing sites json")
+        sites = []
+        if self.sites_path is not None:
+            with open(self.sites_path, "r") as sites_file:
+                sites_json = json.loads(sites_file.read())
+
+            self.information["app"] = self.app.lower()
+
+            self.information["platform"] = {
+                "system": platform.system(),
+                "release": platform.release(),
+                "version": platform.version(),
+                "machine": platform.machine(),
+                "processor": platform.processor(),
+            }
+
+            platform_name = platform.system().lower()
+
+            if isinstance(self.app, AbstractAndroidFirefox):
+
+                device = ADBAndroid()
+
+                for property in ["ro.product.model", "ro.build.user", "ro.build.version.release"]:
+                    self.information[property] = device.shell_output("getprop {}".format(property))
+
+                platform_name = "".join(
+                    e for e in self.information["ro.product.model"] if e.isalnum()
+                )
+
+            for site in sites_json:
+                name = [self.proxy, platform_name, self.app.lower(), site["domain"]]
+                label = site.get("label")
+                if label:
+                    name.append(label)
+                name = "-".join(name)
+
+                site["path"] = os.path.join(self.path, name, name)
+                site["name"] = name
+
+                site["recording_path"] = "%s.mp" % site["path"]
+                site["json_path"] = "%s.json" % site["path"]
+                site["screen_shot_path"] = "%s.png" % site["path"]
+                site["zip_path"] = os.path.join(self.path, "%s.zip" % site["name"])
+                site["manifest_path"] = os.path.join(
+                    self.path, "%s.manifest" % site["name"]
+                )
+
+                sites.append(site)
+        else:
+            raise Exception("No site JSON file found!!!")
```

Comment:
Can you open an issue to clean this up. We should have this verification in the command line parser.

Reply:
https://github.com/mozilla/raptor-studio/issues/20